kong/kong.lua

```diff
@@ -252,6 +252,18 @@ function Kong.balancer()
   end
 end
 
+function Kong.rewrite()
+  core.rewrite.before()
+
+  for _, plugin in ipairs(singletons.loaded_plugins) do
+    if plugin.schema.no_consumer and plugin.schema.no_api then
+      plugin.handler:rewrite()
+    end
+  end
+
+  core.rewrite.after()
+end
+
```

Comment:
`no_api` and `no_consumer` make sure that the plugin cannot be applied to a specific consumer or API, because in the rewrite phase we cannot possibly know who the consumer is and what API he is trying to consume.

With the last commit I am only executing the `rewrite()` check for those plugins that satisfy the requirement.